### PR TITLE
Clippy cleanup for all targets and nighly rust (also support 1.44.0)

### PR DESCRIPTION
**Pull request by svenski123**
#### Problem
Further clippy warning edits, this time for all targets and against rust 1.45.0 (nightly)

#### Summary of Changes
- Numerous edits to address all clippy warnings raised by `rustup run nightly cargo clippy --workspace --all-targets`
- Integer casts from floating point infinity (i.e. from divide by zero) result in zero in rust 1.43 but MAX_INT in 1.45 which broke the bloom filter and rent collector in some test cases; these classes are fixed however this rust language change may cause issues elsewhere (see reference links below)

#### References
- https://github.com/rust-lang/rust/blob/master/RELEASES.md#compatibility-notes
- https://github.com/rust-lang/rust/pull/71269/

### CODE REVIEW

File: cli/src/cli.rs [hunk 1]
```diff
@@ -754,25 +754,18 @@ pub fn parse_command(
         ("airdrop", Some(matches)) => {
             let faucet_port = matches
                 .value_of("faucet_port")
-                .unwrap()
+                .ok_or_else(|| CliError::BadParameter("Missing faucet port".to_string()))?
```
Comment (ryoqun) [inline cli/src/cli.rs:757]: Maybe we just use `...required(true)`? Or is there any reason for not doing so?

File: core/src/rpc_pubsub.rs [hunk 2]
```diff
@@ -921,13 +921,11 @@ mod tests {
         });
 
         // Process votes and check they were notified.
-        // FIX-ME-BETTER-LATER - clone below is required for testcase to pass
-        #[allow(clippy::redundant_clone)]
```
Comment (ryoqun) [inline core/src/rpc_pubsub.rs:925]: @svenski123 Btw, do you know why not using `Arc<_>` solves the past instability of this test?

File: runtime/src/accounts_db.rs [hunk 3]
```diff
@@ -617,14 +617,14 @@ impl AccountsDB {
...
 
-        let no_ancestors = HashMap::new();
         let mut accounts_scan = Measure::start("accounts_scan");
         let accounts_index = self.accounts_index.read().unwrap();
         let pubkeys: Vec<Pubkey> = accounts_index.account_maps.keys().cloned().collect();
         // parallel scan the index.
         let (mut purges, purges_in_root) = pubkeys
             .par_chunks(4096)
             .map(|pubkeys: &[Pubkey]| {
+                let no_ancestors = HashMap::new();
```
[the remark this answers is not in the document]
Comment (svenski123) [reply runtime/src/accounts_db.rs]: Creating empty hash maps involves no allocations and is cheap i.e. push a few quadwords on the stack and continue as opposed to pulling a cache line from another core - though this is insignificant in the scheme of things; the removal of copies from the reduction below is far more signficant.

However creating/passing/checking an empty hashmap for each pubkey is the main problem - I've made &Ancestors into Option<&Ancestors> which avoids the empty hash map entirely.

https://doc.rust-lang.org/beta/std/collections/struct.HashMap.html#method.new

File: core/src/rpc_pubsub.rs (hunk 2 above)
Comment (svenski123) [reply core/src/rpc_pubsub.rs:925]: It's not the Arc per se - it was taking ownership of the only handle to the subscription which was then dropped, closing the channel before anything was read.

The correct parameter passing convention here is that of borrowing, which ensure lifetime is at least that of the function and also avoids the need for the function to know anything about where the subscription is stored i.e. in an arc, a mutex, etc. - it just takes a reference.

File: cli/src/cli.rs (hunk 1 above)
Comment (svenski123) [reply cli/src/cli.rs:757]: There's a default value specified for this parameter on line 2,430 but here on line 757 there's no compile time guarantee that the value of "faucet_port" will not be None hence the error handling - far better for the user to see that error message then have the cli tool panic.

But given the default value, the missing faucet port error should never occur.

File: core/src/rpc_pubsub.rs (hunk 2 above)
Comment (ryoqun) [reply core/src/rpc_pubsub.rs:925]: Oh, I see! That makes sense, then. :)

File: runtime/src/accounts_index.rs [hunk 4]
```diff
@@ -245,7 +248,8 @@ mod tests {
         let key = Keypair::new();
         let index = AccountsIndex::<bool>::default();
         let ancestors = HashMap::new();
-        assert!(index.get(&key.pubkey(), &ancestors).is_none());
+        assert!(index.get(&key.pubkey(), Some(&ancestors)).is_none());
+        assert!(index.get(&key.pubkey(), None).is_none());
```
Comment (ryoqun) [inline runtime/src/accounts_index.rs:252]: Nice for adding another case for your change. :)

File: runtime/src/accounts_db.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (ryoqun) [reply runtime/src/accounts_db.rs]: Overall, your updated changes is definitely better!

btw,

> pulling a cache line from another core

Without no basis, I naively thought these is cheaper than this:

> push a few quadwords on the stack and continue

But, I think you're more confident regarding this. TIL thanks for the tip!

**Comment by ryoqun** [comment]
This pr was the basis for #10585 .

**Pull request merged**